Libraries/JavaScriptAppEngine/System/JSTimers/JSTimersExecution.js

```diff
@@ -108,6 +117,34 @@ var JSTimersExecution = {
     }
   },
 
+  callIdleCallbacks: function(frameTime) {
+    // Arbitrary threshold: 10ms remaining in frame.
+    if (17 - (performanceNow() - frameTime) < 10) {
```

Comment:
Let's call it whenever there is some time left in the frame for now. We can iterate on it and improve the scheduling later. The new React reconciler will definitely be a great test for this and we'll be able to optimize for it.